geode-gfsh/src/distributedTest/java/org/apache/geode/management/internal/cli/commands/RestoreRedundancyCommandDUnitTest.java

```diff
@@ -0,0 +1,618 @@
+import org.apache.geode.cache.RegionShortcut;
+import org.apache.geode.internal.cache.InternalCache;
+import org.apache.geode.internal.cache.PartitionAttributesImpl;
+import org.apache.geode.internal.cache.PartitionedRegion;
+import org.apache.geode.management.internal.cli.util.CommandStringBuilder;
+import org.apache.geode.test.dunit.rules.ClusterStartupRule;
+import org.apache.geode.test.dunit.rules.MemberVM;
+import org.apache.geode.test.junit.assertions.CommandResultAssert;
+import org.apache.geode.test.junit.assertions.InfoResultModelAssert;
+import org.apache.geode.test.junit.rules.GfshCommandRule;
+
+public class RestoreRedundancyCommandDUnitTest {
+  @Rule
+  public ClusterStartupRule cluster = new ClusterStartupRule();
+
+  @Rule
+  public GfshCommandRule gfsh = new GfshCommandRule();
+
+  private int locatorPort;
+  private MemberVM locator;
+  private List<MemberVM> servers;
+  private static final int SERVERS_TO_START = 3;
+  private static final String HIGH_REDUNDANCY_REGION_NAME = "highRedundancy";
+  private static final int HIGH_REDUNDANCY_COPIES = SERVERS_TO_START - 1;
+  private static final String LOW_REDUNDANCY_REGION_NAME = "lowRedundancy";
+  private static final String PARENT_REGION_NAME = "colocatedParent";
+  private static final String CHILD_REGION_NAME = "colocatedChild";
+  private static final int SINGLE_REDUNDANT_COPY = 1;
+  private static final String ZERO_REDUNDANCY_REGION_NAME = "zeroRedundancy";
+
+  @Before
+  public void setUp() throws Exception {
+    servers = new ArrayList<>();
+    locator = cluster.startLocatorVM(0);
+    locatorPort = locator.getPort();
+    IntStream.range(0, SERVERS_TO_START)
+        .forEach(i -> servers.add(cluster.startServerVM(i + 1, locatorPort)));
+    gfsh.connectAndVerify(locator);
+  }
+
+  @Test
+  public void restoreRedundancyWithNoArgumentsRestoresRedundancyForAllRegions() {
+    List<String> regionNames = getAllRegionNames();
+    createAndPopulateRegions(regionNames);
+
+    int numberOfServers = servers.size();
+    regionNames.forEach(region -> locator
+        .waitUntilRegionIsReadyOnExactlyThisManyServers(SEPARATOR + region, numberOfServers));
+
+    String command = new CommandStringBuilder(COMMAND_NAME).getCommandString();
+
+    CommandResultAssert commandResult = gfsh.executeAndAssertThat(command).statusIsSuccess()
+        .hasNoSection(ZERO_REDUNDANCY_SECTION)
+        .hasNoSection(UNDER_REDUNDANCY_SECTION);
+
+    InfoResultModelAssert summary = commandResult.hasInfoSection(SUMMARY_SECTION);
+    summary.hasOutput().contains(ZERO_REDUNDANT_COPIES + 0);
+    summary.hasOutput().contains(PARTIALLY_SATISFIED_REDUNDANCY + 0);
+    summary.hasOutput().contains(FULLY_SATISFIED_REDUNDANCY + regionNames.size());
+
+    InfoResultModelAssert satisfiedSection =
+        commandResult.hasInfoSection(SATISFIED_REDUNDANCY_SECTION);
+    satisfiedSection.hasHeader().isEqualTo(REDUNDANCY_SATISFIED_FOR_REGIONS);
+    satisfiedSection.hasOutput().contains(regionNames);
```

Comment:
This code block is duplicated within the test, it can be extracted to a separate method.